website/content/platform/development/contributing/how-to/add_data_to_existing_endpoint.md

```diff
@@ -0,0 +1,860 @@
+
+`HistoricalEps` is what we will reference in the router when we get there.
+
+### Standard QueryParams
+
+The `HistoricalEpsQueryParams` model defines only one parameter, `symbol`. It includes a validation method for converting the symbol to upper case.
+
+```python
+class HistoricalEpsQueryParams(QueryParams):
+    """Historical EPS Query."""
+
+    symbol: str = Field(description=QUERY_DESCRIPTIONS.get("symbol", ""))
+
+    @field_validator("symbol", mode="before", check_fields=False)
+    @classmethod
+    def upper_symbol(cls, v: str) -> str:
+        """Convert symbol to uppercase."""
+        return v.upper()
+```
+
+We will inherit from this class to create our QueryParams model, specific to AlphaVantage. The model will be named, `AlphaVantageHistoricalEpsQueryParams`. Don't worry about it being too long.
+
+### Standard Data
+
+The `HistoricalEpsData` model defines some fields, with two being mandatory: `date` and `symbol`. It includes a validation method for converting the date from an ISO string to a datetime object.
+
+```python
+class HistoricalEpsData(Data):
+    """Historical EPS Data."""
+
+    date: dateType = Field(default=None, description=DATA_DESCRIPTIONS.get("date", ""))
+    symbol: str = Field(description=DATA_DESCRIPTIONS.get("symbol", ""))
+    announce_time: Optional[str] = Field(
+        default=None, description="Timing of the earnings announcement."
+    )
+    eps_actual: Optional[float] = Field(
+        default=None, description="Actual EPS from the earnings date."
+    )
+    eps_estimated: Optional[float] = Field(
+        default=None, description="Estimated EPS for the earnings date."
+    )
+
+    @field_validator("date", mode="before", check_fields=False)
+    def date_validate(cls, v):  # pylint: disable=E0213
+        """Return formatted datetime."""
+        return parser.isoparse(str(v))
+```
+
+Now we know exactly what is going to be added, and how we should structure our query to fetch the data. This endpoint is not overly complex but harmonizing many input parameters, and the potentially endless data fields, across many providers is a challenge.
+
+So far, we have knocked out three of the outlined tasks.
+
+  - [X] Catalogue the parameters and returned fields from the chosen data provider.
+  - [X] Find the existing standard model that is mapped to the router endpoint.
+  - [X] Identify common parameters and fields to map.
+
+Let's get on with the fun stuff and start building!
+
+## Build the Provider Model
+
+### Create a New File
+
+The first step is to create a new file in the provider extension folder:
+
+```console
+~/OpenBBTerminal/openbb_platform/providers/alpha_vantage/openbb_alpha_vantage/models
+```
+
+We will call this file: `historical_eps.py`
+
+The first line of the file should be a docstring, followed by the import statements.
```

Comment:
This should be caught by our pre-commit hooks. Maybe we should ask people to install the dev extension that will in turn have these hooks?